qiskit_experiments/warnings.py

```diff
@@ -0,0 +1,245 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""A collection of typical warnings."""
+
+import functools
+import warnings
+
+from typing import Callable, Optional, Type, Dict
+
+
+def deprecated_logic(
```

Comment:
```suggestion
def deprecated_function(
```
I find logic an odd choice for name, I would just call this deprecated function (methods are still functions)

This should also probably take a stacklevel arg like terra `deprecation_function` method.